Sources/NIOAsyncAwaitDemo/AsyncChannelIO.swift

```diff
@@ -15,8 +15,8 @@
 import NIOCore
 import NIOHTTP1
 
-#if compiler(>=5.5) && canImport(_Concurrency)
-@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+#if canImport(_Concurrency) && compiler(>=5.5.2) || (!os(macOS) && compiler(>=5.5))
```

Comment:
The `os` guards here are insufficient to cover the use-case (we'd need to check for all Apple platforms), but I don't think it matters. On non-Apple platforms the two function implementations are totally fine, so we can just drop the `||` here.

Reply:
Good point, I have updated the PR to incorporate your feedback 👍